client/db/src/lib.rs

```diff
@@ -1345,8 +1345,7 @@ impl<Block: BlockT> Backend<Block> {
 
 			let parent_hash = *pending_block.header.parent_hash();
 			let number = *pending_block.header.number();
-			let existing_header =
-				number <= best_num && self.blockchain.header(BlockId::hash(hash))?.is_some();
+			let existing_header = self.blockchain.header(BlockId::hash(hash))?.is_some();
```

Comment:
test at line 1561 becomes redundant.

Reply:
Do you mean this?

https://github.com/paritytech/substrate/blob/b8706b4e28709631387b74a2ce846f5fa71e3127/client/db/src/lib.rs#L1561-L1563

If yes, here we're testing the `parent_hash` and not `hash`

Reply:
Oh sure you're right.